src/T2DMap.cpp

```diff
@@ -215,6 +215,21 @@ QColor T2DMap::getColor(int id)
         break;
     default: //user defined room color
         if (!mpMap->customEnvColors.contains(env)) {
+            if (16 < env < 232)
+            {
+                int base = env - 16;
+                int r = base / 36;
+                int g = (base - (r * 36)) / 6;
+                int b = (base - (r * 36)) - (g * 6);
```

Comment:
No need to do the multiplications separately, just do:
```C
                quint8 r = 51 * (base / 36);
                quint8 g = 51 * ((base - (r * 36)) / 6);
                quint8 b = 51 * ((base - (r * 36)) - (g * 6));
```

Reply:
Changed